spec/02-integration/05-proxy/29-stream_upstream_tls_spec.lua

```diff
@@ -0,0 +1,527 @@
+local helpers = require "spec.helpers"
```

Comment:
There's a lot of duplication between this and `spec/02-integration/05-proxy/18-upstream_tls_spec.lua`. I think we should combine them if feasible. After your PR, the upstream TLS features will be fairly symmetric between `http` and `stream` subsystems (:trophy:), so having the tests "live" alongside each other in the same file decreases the odds that someone will add a new upstream TLS feature and forget to test it with a `tls` service or vice versa.

With just a little bit of refactoring, I think this could all be done pretty cleanly:

```lua

local SUBSYSTEMS = {
  http = {
    service_protocol = "https",
    hostname = "https.test",
  },
  stream = {
    service_protocol = "tls",
    hostname = "tls.test",
  }
}

-- ...

lazy_setup(function()

  -- ...

  for subsystem, conf in pairs(SUBSYSTEMS) do
    -- setup both https/tls services/routes/upstreams

  end

end)

for subsystem, conf in pairs(SUBSYSTEMS) do
  describe("mtls feature blah blah with " .. conf.service_protocol .. " service", function()
    -- ...
  end)
end
```